streams/examples/src/main/java/org/apache/kafka/streams/examples/temperature/TemperatureDemo.java

```diff
@@ -0,0 +1,109 @@
+ *
+ * Before running this example you must create the input topic and the output topic (e.g. via
+ * bin/kafka-topics.sh --create ...), and write some data to the input topic (e.g. via
+ * bin/kafka-console-producer.sh). Otherwise you won't see any data arriving in the output topic.
+ */
+public class TemperatureDemo {
+
+    public static void main(String[] args) throws Exception {
+
+        Properties props = new Properties();
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-temperature");
+        props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
+        props.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+        props.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        props.put(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, 0);
+
+        KStreamBuilder builder = new KStreamBuilder();
+
+        KStream<String, String> source = builder.stream("temperature");
+
+        KStream<Windowed<String>, String> max = source
+                .map(new KeyValueMapper<String, String, KeyValue<String, String>>() {
+                    @Override
+                    public KeyValue<String, String> apply(String key, String value) {
+                        return new KeyValue<>("temp", value);
+                    }
+                })
+                .groupByKey()
+                .reduce(new Reducer<String>() {
+                    @Override
+                    public String apply(String value1, String value2) {
+                        if (Integer.parseInt(value1) > Integer.parseInt(value2))
+                            return value1;
+                        else
+                            return value2;
+                    }
+                }, TimeWindows.of(5000))
+                .toStream()
+                .filter(new Predicate<Windowed<String>, String>() {
+                    @Override
+                    public boolean test(Windowed<String> key, String value) {
+                        return Integer.parseInt(value) > 20;
+                    }
+                });
+
+        WindowedSerializer<String> windowedSerializer = new WindowedSerializer<>(Serdes.String().serializer());
+        WindowedDeserializer<String> windowedDeserializer = new WindowedDeserializer<>(Serdes.String().deserializer());
+        Serde<Windowed<String>> windowedSerde = Serdes.serdeFrom(windowedSerializer, windowedDeserializer);
+
+        // need to override key serde to Windowed<String> type
+        max.to(windowedSerde, Serdes.String(), "max");
+
+        KafkaStreams streams = new KafkaStreams(builder, props);
+        streams.start();
+
+        // usually the stream application would be running forever,
+        // in this example we just let it run for some time (big enough for covering the used window) and then stop
+        Thread.sleep(30000);
```

Comment:
How about changing this to be only stoppable by ctrl-C? We are changing the rest of the examples as well in a manner to improve our quick start: https://github.com/apache/kafka/pull/3515